### DISCUSSION TITLE
Client side data fetching

### DISCUSSION THREAD

**Discussion by zachrip** [vercel/swr#1622] [Q&A]
To preface: I'm new to next.js and swr.

In the [client side data fetching section](https://swr.vercel.app/docs/with-nextjs#client-side-data-fetching) it says to "First, immediately show the page without data. You can show loading states for missing data. Then, fetch the data on the client side and display it when ready."

When attempting to do this, swr is still making a call to the backend from the server side, which is never used because it's not waited for by next. This doesn't seem right to me. I know I can disable ssr for components but that doesn't seem to be what the docs are asking me to do. Am I doing something wrong? Can someone show me an example of exactly what the docs are describing?

Thanks!

**Comment by samcx** [comment]
Is there a minimal reproduction repository you can share with us so we can see where you are using your swr code?

**Reply by zachrip** [reply]
Actually the examples illustrate this. If you clone the basic-typescript example and add a console log to the API handler, you'll see it executes twice for each request. But the server doesn't actually use the data.

**Reply by samcx** [reply]
Is this happening in the Production build as well?

**Reply by zachrip** [reply]
I haven't tested that but I would imagine so. I can try it.

**Reply by zachrip** [reply]
I tested with revalidateOnFocus disabled and the problem went away - I should've realized sooner but what I was seeing was nextjs hotloading my updates which caused an api request, then swr revalidating when I went to the browser.

@samsisle feel free to remove this discussion, I have my answers 😃

**Reply by spavlovic77** [reply]
> I tested with revalidateOnFocus disabled and the problem went away - I should've realized sooner but what I was seeing was nextjs hotloading my updates which caused an api request, then swr revalidating when I went to the browser.
> 
> @samsisle feel free to remove this discussion, I have my answers 😃

@zachrip  I have similar issue with my Dependent Fetching. The data is in Console but not rendered on screen. Can you please have a look in my code, it this is the same problem which you have had?

const { data: stakedByStaker } = useSWR('stakedByStaker', ()=> fetcherStakedByStaker(slice, accounts))
const { data: listStaked, error } = useSWR(()=>'listStaked'+stakedByStaker[0].staker, ()=>fetcherStakedFights(stakedByStaker, accounts, web3))

if (error) return <div>failed to load</div>
if (!listStaked) return <div>loading...</div>
return (...)

**Reply by samcx** [reply]
@zachrip I'm glad you were able to figure it out! I remembered going through the same thought process as you, but I didn't have an answer ready for this.

I just remembered that I wasn't seeing the behavior in Production.
